packages/name-controller/src/NameController.ts

```diff
@@ -153,12 +170,21 @@ export class NameController extends BaseController<
   setName(request: SetNameRequest) {
     this.#validateSetNameRequest(request);
 
-    const { value, type, name, sourceId: requestSourceId, variation } = request;
+    const {
+      value,
+      type,
+      name,
+      sourceId: requestSourceId,
+      origin: requestOrigin,
+      variation,
+    } = request;
     const sourceId = requestSourceId ?? null;
+    const origin = requestOrigin ?? null;
 
     this.#updateEntry(value, type, variation, (entry: NameEntry) => {
       entry.name = name;
       entry.sourceId = sourceId;
+      entry.origin = origin ?? NameOrigin.API;
```

Comment:
Could we set the `API` fallback on line `182` to avoid this additional ternary and just assign to `origin` here?